pkg/util/procfs/procfs.go

```diff
@@ -62,7 +65,42 @@ func (pfs *ProcFS) GetFullContainerName(pid int) (string, error) {
 	return containerNameFromProcCgroup(string(content))
 }
 
-func PidOf(name string) []int {
+func PKill(name string, sig syscall.Signal) error {
+	if len(name) == 0 {
+		return fmt.Errorf("name should not be empty")
+	}
+	re, err := regexp.Compile(name)
+	if err != nil {
+		return err
+	}
+	pids := getPids(re)
+	if len(pids) == 0 {
+		return fmt.Errorf("unable to fetch pids for process name : %q", name)
+	}
+	errList := []error{}
+	for _, pid := range pids {
+		if err = syscall.Kill(pid, sig); err != nil {
+			errList = append(errList, err)
+		}
+	}
+	if len(errList) > 0 {
+		return utilerrors.NewAggregate(errList)
```

Comment:
nit: just return `utilerrors.NewAggregate(errList)`. `NewAggregate` returns `nil` if `errList` is empty.